client/src/components/VisibilityControl/VisibilityControl.tsx

```diff
@@ -22,7 +39,9 @@ export class VisibilityControl extends React.Component {
     return (
       <div className="visibility-control">
         {_.map(items, (item) => {
-          const item_components = {
+          const item_components: {
+            [key: string]: boolean | JSX.Element | Element;
```

Comment:
`[key: "count" | "icon" | "text"]: React.ReactNode`

If you've addressed my first comment then `"count" | "icon" | "text"` is now duplicated, so probably extract that to it's own type to DRY things out.